### PR TITLE
store: add missing case for list-keys

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

https://bugs.launchpad.net/snapcraft/+bug/1996618

CRAFT-1804

### CODE REVIEW

**Comment by lengau** [comment]
The linting issue is caused by https://github.com/microsoft/pyright/issues/5394

I've worked around it in https://github.com/snapcore/snapcraft/pull/4233

File: snapcraft_legacy/_store.py [hunk 1]
```diff
@@ -439,34 +439,45 @@ def list_keys():
...
+                headers=["", "Name", "SHA3-384 fingerprint", ""],
+                tablefmt="plain",
+            )
+            print(
+                "No keys have been registered with this account.\n"
+                "The following keys are available on this system:"
+            )
+            print(tabulated_keys)
+        else:
+            print("No keys have been registered.")
```
Comment (pedronis) [inline snapcraft_legacy/_store.py:479]: it seems that this path should print:

"No keys have been created on this system. "
" See 'snapcraft create-key --help' to create a key."

as well

**Review by pedronis — COMMENTED** [review]
this looks conceptually correct. Somebody in snapcraft should review the tests and UI aspects of this. One comment about the no keys of any kind case.

File: snapcraft_legacy/_store.py (hunk 1 above)
Comment (lengau) [reply snapcraft_legacy/_store.py:479]: Agreed! I've changed it up a bit so now it always hints about registering keys if none are registered and about creating keys if none are available on the system (effectively decoupling them except when displaying registered keys that are unavailable on the system). The constants at the top of `test_keys.py` give an example of how those might look.

**Review by mr-cal — APPROVED** [review]

File: snapcraft_legacy/_store.py [hunk 2]
```diff
@@ -453,19 +453,29 @@ def list_keys():
...
         print(
             "No keys have been created on this system. "
-            " See 'snapcraft create-key --help' to create a key.\n"
-            "The following SHA3-384 key fingerprints have been registered "
-            f"but are not available on this system:\n{registered_keys}"
+            "See 'snapcraft create-key --help' to create a key."
         )
+    if enabled_keys:
+        local_hashes = {key["sha3-384"] for key in keys}
+        registered_keys = "\n".join(
```
Comment (pedronis) [inline snapcraft_legacy/_store.py:467]: unavailable_registered_keys might be a clearer name for this?

**Review by pedronis — APPROVED** [review]
+1

**Comment by sergiusens** [comment]
the hotfix is the fix that will later be merged back to main

**Pull request closed without merging**